### PR TITLE
Add markdown link checker to pre-commit hooks

### PR DESCRIPTION
**Pull request by m-kuhn**
State of affairs:

<pre><span style="color:#C01C28">  ERROR: 1 dead links found in documentation/get-started/faq.en.md !</span>
  [<span style="color:#C01C28">✖</span>] https://qfield.cloud/faq/ → Status: 404

<span style="color:#C01C28">  ERROR: 1 dead links found in documentation/get-started/third-part-tutorials.en.md !</span>
  [<span style="color:#C01C28">✖</span>] https://www.udemy.com/course/qfield-quickstart/ → Status: 403

<span style="color:#C01C28">  ERROR: 1 dead links found in documentation/success-stories/river-state-survey.en.md !</span>
  [<span style="color:#C01C28">✖</span>] http://http://www.milvignes.ch/ → Status: 0

<span style="color:#C01C28">  ERROR: 1 dead links found in documentation/how-to/attributes-form.en.md !</span>
  [<span style="color:#C01C28">✖</span>] ../../get-started/support/ → Status: 400

<span style="color:#C01C28">  ERROR: 2 dead links found in documentation/reference/qfieldcloud/jobs.en.md !</span>
  [<span style="color:#C01C28">✖</span>] projects.md → Status: 400
  [<span style="color:#C01C28">✖</span>] specs.md → Status: 400

<span style="color:#C01C28">  ERROR: 3 dead links found in documentation/how-to/gnss.en.md !</span>
  [<span style="color:#C01C28">✖</span>] ../digitize/ → Status: 400
  [<span style="color:#C01C28">✖</span>] ../gnss/ → Status: 400
  [<span style="color:#C01C28">✖</span>] https://docs.qgis.org/latest/en/docs/user_manual/working_with_vector/vector_properties.html → Status: 0

<span style="color:#C01C28">  ERROR: 1 dead links found in documentation/how-to/standalone-datasets.en.md !</span>
  [<span style="color:#C01C28">✖</span>] /get-started/storage/ → Status: 400

<span style="color:#C01C28">  ERROR: 3 dead links found in documentation/get-started/contribute.en.md !</span>
  [<span style="color:#C01C28">✖</span>] https://twitter.com/share?text=Looking%20for%20a%20good%20tool%20for%20field%20work%20in%20GIS?%20Check%20out%20%23QField! → Status: 0
  [<span style="color:#C01C28">✖</span>] https://twitter.com/QFieldForQGIS → Status: 0
  [<span style="color:#C01C28">✖</span>] https://github.com/opengisch/QField/blob/master/doc/dev.en.md → Status: 404

<span style="color:#C01C28">  ERROR: 1 dead links found in documentation/get-started/sponsor.en.md !</span>
  [<span style="color:#C01C28">✖</span>] https://www.opengis.ch/training-consulting/ → Status: 0

<span style="color:#C01C28">  ERROR: 2 dead links found in documentation/how-to/projects.en.md !</span>
  [<span style="color:#C01C28">✖</span>] ../../get-started/tutorials/get-started-qfc/ → Status: 400
  [<span style="color:#C01C28">✖</span>] ../../get-started/storage-en/ → Status: 400

<span style="color:#C01C28">  ERROR: 1 dead links found in documentation/reference/data-format.en.md !</span>
  [<span style="color:#C01C28">✖</span>] https://github.com/opengisch/QField-docs/edit/master/en/project-management/dataformat.rst → Status: 404

<span style="color:#C01C28">  ERROR: 1 dead links found in documentation/get-started/tutorials/get-started-qfs.en.md !</span>
  [<span style="color:#C01C28">✖</span>] ../picture/ → Status: 400

<span style="color:#C01C28">  ERROR: 1 dead links found in documentation/how-to/pictures.en.md !</span>
  [<span style="color:#C01C28">✖</span>] ../attributes-form.md → Status: 400
</pre>

### CODE REVIEW

**Comment by lucienicolier** [comment]
@m-kuhn The links that are reported seem to work. Tested locally build and on the currant doc.

File: documentation/get-started/tutorials/get-started-qfc.en.md [hunk 1]
```diff
@@ -117,7 +117,7 @@ A form will ask you for project name, description and local directory. In the lo
 
 ### Configure your project layers for QField
 
-Configure the project layers by clicking the fifth icon in the QFieldSync toolbar ![](../../assets/images/project_properties.svg){Width="20px"}. Here you can configure QFieldCloud layer actions. Most of the time you need to configure a preference either to online or offline layers. For more fine grained control, in the advanced settings you can configure the action layer by layer. Get more information about how to configure your layers in the [Get Started guide for QFieldSync](./get-started-qfs.md).
+Configure the project layers by clicking the fifth icon in the QFieldSync toolbar ![](../../assets/images/project_properties.svg){Width="20px"}. Here you can configure QFieldCloud layer actions. Most of the time you need to configure a preference either to online or offline layers. For more fine grained control, in the advanced settings you can configure the action layer by layer. Get more information about how to configure your layers in the [Get Started guide for QFieldSync](./get-started-qfs.md)!.
```
Comment (m-kuhn) [inline documentation/get-started/tutorials/get-started-qfc.en.md:120]: I think `.en` is missing ....

Comment (m-kuhn) [reply documentation/get-started/tutorials/get-started-qfc.en.md:120]: you are right, it's not the problem.

**Comment by m-kuhn** [comment]
@lucienicolier for external links (like milvignes.ch) you can use this

https://github.com/opengisch/QField-docs/pull/340/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R117

```
<!-- markdown-link-check-disable-line -->
```

**Comment by m-kuhn** [comment]
I fixed some of the checks (sorry they were not ok yet ...)

I would suggest we always use links to `[file].md` from now on (no `file/` in the end) and ignore all external links with `<!-- markdown-link-check-disable-line -->`

File: documentation/how-to/gnss.en.md [hunk 2]
```diff
@@ -164,7 +164,8 @@ When active, holding the add vertex button is not required, a short tap on the b
 
 !![](../assets/images/positioning_averaged_set.jpg)
 
-When using [`@gnss_*` or `@position_` variables](../gnss/#positioning-variables) on averaged positions, the variable will also represent the average over all collected samples.
+When using [`@gnss_*` or `@position_` variables](../gnss.md#positioning-variables) on averaged positions, the variable will also represent the average over all collected samples.
```
Comment (m-kuhn) [inline documentation/how-to/gnss.en.md:167]:
```suggestion
When using [`@gnss_*` or `@position_` variables](./gnss.md#positioning-variables) on averaged positions, the variable will also represent the average over all collected samples.
```

File: documentation/how-to/attributes-form.en.md [hunk 3]
```diff
@@ -23,7 +23,7 @@ The field widgets from QGIS are supported on a best effort basis and optimized f
...
 | UUID Generator     | :material-check: |                                                                                                                                                                                                        |
 | QML / HTML Widget  | :material-check: |                                                                                                                                                                                                        |
-| Others             | :material-close: | [Funding](../../get-started/support.md#development-and-custom-apps)                                                                                                                                      |
+| Others             | :material-close: | [Funding](././get-started/support.md#development-and-custom-apps)                                                                                                                                      |
```
Comment (m-kuhn) [inline documentation/how-to/attributes-form.en.md:26]:
```suggestion
| Others             | :material-close: | [Funding](./get-started/support.md#development-and-custom-apps)                                                                                                                                      |
```

@lucienicolier `./` means "the current directory", repeating it doesn't have any influence (maybe we could even skip it completely here)

**Comment by lucienicolier** [comment]
@m-kuhn ;) 
Can I merge it?

**Review by lucienicolier — APPROVED** [review]

**Pull request merged**